networkx/algorithms/centrality/load.py

```diff
@@ -108,91 +110,107 @@ def _node_betweenness(G,source,cutoff=False,normalized=True,weight=None):
+    This function is for demonstration and testing purposes.
+
+    Parameters
+    ----------
+    G : graph
+        A networkx graph
 
-    This module is for demonstration and testing purposes.
+    cutoff : bool, optional
+        If specified, only consider paths of length <= cutoff.
 
+    Returns
+    -------
+    A dict keyed by edge 2-tuple to the number of shortest paths
+    which use that edge. Where more than one path is shortest
+    the count is divided equally among paths.
     """
-    betweenness={}
-    if not nodes:         # find betweenness for every node  in graph
-        nodes = list(G)   # that probably is what you want...
-    for source in nodes:
-        ubetween=_edge_betweenness(G,source,nodes,cutoff=cutoff)
-        for v in ubetween.keys():
-            b=betweenness.setdefault(v,0)  # get or set default
-            betweenness[v]=ubetween[v]+b    # cumulative total
+    betweenness = {}
+    for source in G:
+        ubetween = _edge_betweenness(G, source, cutoff=cutoff)
+        for e, ubetweenv in ubetween.items():
+            if e in betweenness:
+                betweenness[e] += ubetweenv  # cumulative total
+            else:
+                betweenness[e] = ubetweenv
     return betweenness
 
-def _edge_betweenness(G,source,nodes,cutoff=False):
+
+def _edge_betweenness(G, source, nodes=None, cutoff=False):
     """
     Edge betweenness helper.
     """
-    between={}
     # get the predecessor data
-    #(pred,length)=_fast_predecessor(G,source,cutoff=cutoff)
-    (pred,length)=nx.predecessor(G,source,cutoff=cutoff,return_seen=True)
+    (pred, length) = nx.predecessor(G, source, cutoff=cutoff, return_seen=True)
     # order the nodes by path length
-    onodes = [ nn for dd,nn in sorted( (dist,n) for n,dist in length.items() )]
+    onodes = [n for n, d in sorted(length.items(), key=itemgetter(1))]
     # intialize betweenness, doesn't account for any edge weights
-    for u,v in G.edges(nodes):
-        between[(u,v)]=1.0
-        between[(v,u)]=1.0
+    between = {}
+    for u, v in G.edges(nodes):
+        between[(u, v)] = 1.0
+        between[(v, u)] = 1.0
 
     while onodes:           # work through all paths
-        v=onodes.pop()
+        v = onodes.pop()
         if v in pred:
-            num_paths=len(pred[v])   # Discount betweenness if more than
-            for w in pred[v]:        # one shortest path.
+            # Discount betweenness if more than one shortest path.
+            num_paths = len(pred[v])
+            for w in pred[v]:
                 if w in pred:
-                    num_paths=len(pred[w])  # Discount betweenness, mult path
+                    # Discount betweenness, mult path
+                    num_paths = len(pred[w])
                     for x in pred[w]:
-                        between[(w,x)]+=between[(v,w)]/num_paths
-                        between[(x,w)]+=between[(w,v)]/num_paths
+                        between[(w, x)] += between[(v, w)] / num_paths
```

Comment:
This module should have `from __future__ import division` if it includes division.